picoquic/tls_api.c

```diff
@@ -641,8 +654,10 @@ void picoquic_master_tlscontext_free(picoquic_quic_t* quic)
             ctx->get_time = NULL;
         }
 
-        if (ctx->certificates.list != NULL) {
-            free(ctx->certificates.list);
+        free_certificates_list(ctx->certificates.list, ctx->certificates.count);
+
+        if (ctx->sign_certificate != NULL) {
+            ptls_openssl_dispose_sign_certificate((ptls_openssl_sign_certificate_t*)ctx->sign_certificate);
         }
 
```

Comment:
Did we in fact have a memory leak before? Did you find out with a tool, or just by code review?

Reply:
Just by code review.

Reply:
And yes, I assume that before that would be a memory leak.

Reply:
Do you happen to know a good open source tool to detect leaks?

Reply:
Yep, `valgrind` is the tool to use :)